api/v1beta1/cloudstackcluster_webhook.go

```diff
@@ -96,8 +96,12 @@ func (r *CloudStackCluster) ValidateUpdate(old runtime.Object) error {
 	// No spec fields may be changed
 	errorList = webhook_utilities.EnsureStringFieldsAreEqual(spec.Zone, oldSpec.Zone, "zone", errorList)
 	errorList = webhook_utilities.EnsureStringFieldsAreEqual(spec.Network, oldSpec.Network, "network", errorList)
-	errorList = webhook_utilities.EnsureStringFieldsAreEqual(spec.ControlPlaneEndpoint.Host, oldSpec.ControlPlaneEndpoint.Host, "controlplaneendpointhost", errorList)
-	errorList = webhook_utilities.EnsureStringFieldsAreEqual(string(spec.ControlPlaneEndpoint.Port), string(oldSpec.ControlPlaneEndpoint.Port), "controlplaneendpointport", errorList)
+	if oldSpec.ControlPlaneEndpoint.Host != "" { // Need to allow one time endpoint setting via CAPC cluster controller.
+		errorList = webhook_utilities.EnsureStringFieldsAreEqual(
+			spec.ControlPlaneEndpoint.Host, oldSpec.ControlPlaneEndpoint.Host, "controlplaneendpointhost", errorList)
+		errorList = webhook_utilities.EnsureStringFieldsAreEqual(
+			string(spec.ControlPlaneEndpoint.Port), string(oldSpec.ControlPlaneEndpoint.Port), "controlplaneendpointport", errorList)
+	}
```

Comment:
This is a fixup for the scenario in which the host or port is not specified at cluster creation time.

The cluster controller needs to be able to set this.

Reply:
why wouldn't host/port be specified at cluster creation time?

Reply:
This is just the way it was designed originally. We can change it if we like. It's use case is just grabbing a random IP for the endpoint that's available on the specified network.